lib/Sema/CSSimplify.cpp

```diff
@@ -4846,6 +4871,17 @@ ConstraintSystem::simplifyFixConstraint(Fix fix, Type type1, Type type2,
 
     return result;
   }
+
+  case FixKind::UnwrapOptionalBase: {
+    // First type already appropriately set.
+    auto result = matchTypes(type1, type2, matchKind, subflags, locator);
+    if (result == SolutionKind::Solved)
+      if (recordFix(fix, locator))
```

Comment:
I suggest that you `recordFix()` first, then directly return the result of the call to `matchTypes`. That will handle the "unresolved" case appropriately, rather than losing the fix.